src/app/core/auth/state/auth-state.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Action, State, StateContext } from '@ngxs/store'
+import { Injectable } from '@angular/core'
+import { AuthService } from 'src/app/core/auth/services/auth.service'
+import { Observable, tap } from 'rxjs'
+import { Tokens } from 'src/app/core/auth/types/tokens'
+import { TokensStorageService } from "../services/tokens-storage.service";
+import { Login, LoginOut, SetTokens, UpdateLoginFormSubmission } from "./auth.actions";
+import { Router } from "@angular/router";
+
+export interface AuthStateModel {
```

Comment:
Do not add suffix/prefix to entity names (e.g. no `Model`, `I`, `Interface`, etc.)